File: 26_removeDuplicates.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if len(nums) == 0:
            return 0
        insert = 0
        for i in range(len(nums)-1):
            if nums[i] == nums[i+1]:
                continue
            else:
                nums[insert] = nums[i]
                insert += 1
        nums[insert] = nums[-1]
        return insert+1

File: 26_removeDuplicates.c
int removeDuplicates(int* nums, int numsSize){
    
    if (!numsSize) {
        return 0;
    }
    
    int j = 0;
    int i = 0;
    
    while (i < numsSize-1) {
        if (nums[i] != nums[i+1]) {
            nums[j++] = nums[i];
        }
        i++;
    }
    nums[j] = nums[i];
    return j+1;
}
